### PR TITLE
Added Batch AI support

### PR DESCRIPTION
**Pull request by lenala**

### CODE REVIEW

**Comment by hovsepm** [comment]
>     <PackageReference Include="Microsoft.Azure.Management.Batch.Fluent" Version="1.4.1" />

Please add BatchAI package dependency

---
Refers to: src/ResourceManagement/Azure.Fluent/Microsoft.Azure.Management.Fluent.csproj:34 in c182454. [](commit_id = c1824548d586b86160961af8ee4b6249cb1615e1, deletion_comment = False)

File: src/ResourceManagement/BatchAI/BatchAIClusterImpl.cs [hunk 1]
```diff
@@ -0,0 +1,360 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Linq;
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/BatchAIClusterImpl.cs]: Please fix the document formatting. Ctr+A then Ctr+K,D.

File: src/ResourceManagement/BatchAI/BatchAIJobsImpl.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            return await Inner.GetAsync(resourceGroupName, name);
+        }
+
+        protected override BatchAIJobImpl WrapModel(string name)
+        {
+            return new BatchAIJobImpl(name, parent, new JobInner());
+//            {
+//                Region = parent.Region,
+//            }
+//            .WithRegion(parent.RegionName).WithExistingResourceGroup(parent.ResourceGroupName);
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/BatchAIJobsImpl.cs]: Please remove this comment.

File: src/ResourceManagement/BatchAI/Domain/IBatchAICluster.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        /// Gets All virtual machines in a cluster are the same size. For information
+        /// about available VM sizes for clusters using images from the Virtual
+        /// Machines Marketplace (see Sizes for Virtual Machines (Linux) or Sizes
+        /// for Virtual Machines (Windows). Batch AI service supports all Azure VM
+        /// sizes except STANDARD_A0 and those with premium storage (STANDARD_GS,
+        /// STANDARD_DS, and STANDARD_DSV2 series).
+        /// </summary>
+        /// <summary>
+        /// Gets the size of the virtual machines in the cluster.
+        /// </summary>
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/Domain/IBatchAICluster.cs]: Please check the documentation - why do we have 2 summary blocks?

File: src/ResourceManagement/BatchAI/Domain/IBatchAICluster.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        string VMSize { get; }
+
+        /// <summary>
+        /// Gets The default value is dedicated. The node can get preempted while the
+        /// task is running if lowpriority is choosen. This is best suited if the
+        /// workload is checkpointing and can be restarted.
+        /// </summary>
+        /// <summary>
+        /// Gets virtual machine priority status.
+        /// </summary>
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/Domain/IBatchAICluster.cs]: Same here why do we have 2 summery blocks?

File: src/ResourceManagement/BatchAI/Domain/IBatchAIJob.cs [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        /// executing. terminating - The job is terminated by the user, the
+        /// terminate operation is in progress. succeeded - The job has completed
+        /// running succesfully and exited with exit code 0. failed - The job has
+        /// finished unsuccessfully (failed with a non-zero exit code) and has
+        /// exhausted its retry limit. A job is also marked as failed if an error
+        /// occurred launching the job.
+        /// </summary>
+        /// <summary>
+        /// Gets the current state of the job.
+        /// </summary>
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/Domain/IBatchAIJob.cs]: same comment - second summary can be removed, right?

File: src/ResourceManagement/BatchAI/Domain/BatchAICluster/Definition/IDefinition.cs [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    /// Defines the volumes to mount on the cluster.
+    /// </summary>
+    public interface IWithMountVolumes
+    {
+        /// <summary>
+        /// Gets Begins the definition of Azure file share reference to be mounted on each cluster node.
+        /// </summary>
+        /// <summary>
+        /// Gets the first stage of file share reference definition.
+        /// </summary>
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/Domain/BatchAICluster/Definition/IDefinition.cs]: please fix the comments.

File: src/ResourceManagement/BatchAI/Domain/BatchAICluster/Definition/IDefinition.cs [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+        AzureFileShare.Definition.IBlank<Microsoft.Azure.Management.BatchAI.Fluent.BatchAICluster.Definition.IWithCreate> DefineAzureFileShare();
+
+        /// <summary>
+        /// Gets Begins the definition of Azure blob file system reference to be mounted on each cluster node.
+        /// </summary>
+        /// <summary>
+        /// Gets the first stage of Azure blob file system reference definition.
+        /// </summary>
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/Domain/BatchAICluster/Definition/IDefinition.cs]: please fix the comments

File: src/ResourceManagement/BatchAI/Domain/BatchAICluster/Definition/IDefinition.cs [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+       AzureBlobFileSystem.Definition.IBlank<Microsoft.Azure.Management.BatchAI.Fluent.BatchAICluster.Definition.IWithCreate> DefineAzureBlobFileSystem();
+
+        /// <summary>
+        /// Gets Begins the definition of Azure file server reference.
+        /// </summary>
+        /// <summary>
+        /// Gets the first stage of file server reference definition.
+        /// </summary>
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/Domain/BatchAICluster/Definition/IDefinition.cs]: comments

File: src/ResourceManagement/BatchAI/Domain/BatchAICluster/Definition/IDefinition.cs [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+        FileServer.Definition.IBlank<Microsoft.Azure.Management.BatchAI.Fluent.BatchAICluster.Definition.IWithCreate> DefineFileServer();
+
+        /// <summary>
+        /// Specifies the details of the file system to mount on the compute cluster nodes.
+        /// </summary>
+        /// <param name="mountCommand">Command used to mount the unmanaged file system.</param>
+        /// <param name="relativeMountPath">The relative path on the compute cluster node where the file system will be mounted.</param>
+        /// <return>The next stage of Batch AI cluster definition.</return>
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/Domain/BatchAICluster/Definition/IDefinition.cs]: comments.

File: src/ResourceManagement/BatchAI/BatchAIClusterImpl.cs (hunk 1 above)
Comment (lenala) [reply src/ResourceManagement/BatchAI/BatchAIClusterImpl.cs]: Fixed, and for other classes too.

**Pull request merged**